### PR TITLE
GCP Auth extension - allow users to specify quota project ID

### LINKED ISSUE #1645: GCP Auth Extension - allow users to explicitly specify quota project ID
### Component(s)

gcp-auth-extension

### Is your feature request related to a problem? Please describe.

The GCP Auth extension currently relies only on Application Default Credentials to infer the Quota project ID to be used for the export. This means users cannot explicitly specify a quota project ID that should be used for the export.

### Describe the solution you'd like

The feature request is for following changes:
 - Expose an optional argument for the users to explicitly specify quota project ID to be used.
    - If the argument is not specified, it is inferred from the Application Default Credentials.
    - If the resolved value of the quota project ID is empty or null, the quota user project header: `X-Goog-User-Project` will not be added. 
 - Make `GOOGLE_CLOUD_PROJECT` required option.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by psx95**
**Description:**

This PR provides an optional argument that lets the users specify a quota project ID.

**Existing Issue(s):**

Fixes #1645 

**Testing:**
 - Unit tests verifying `google.cloud.project` as required field added.
 - Manually tested the extension JAR with test applications.

**Documentation:**

Updates to the README reflecting the user facing changes added.

**Outstanding items:**

Add parameterized unit tests to ensure behavior around null or empty values for the project IDs.

### CODE REVIEW

File: gcp-auth-extension/src/test/java/io/opentelemetry/contrib/gcp/auth/GcpAuthAutoConfigurationCustomizerProviderTest.java
Comment (aabmass) [inline gcp-auth-extension/src/test/java/io/opentelemetry/contrib/gcp/auth/GcpAuthAutoConfigurationCustomizerProviderTest.java:1]: IIUC there was a bug before that we would send an empty (null?) quota project header if it wasn't configured which made requests fail.

Is there a test that confirms we don't add the quota project header if it is not configured?

File: gcp-auth-extension/README.md [hunk 1]
```diff
@@ -36,7 +36,10 @@ Here is a list of configurable options for the extension:
 
 - `GOOGLE_CLOUD_PROJECT`: Environment variable that represents the Google Cloud Project ID to which the telemetry needs to be exported.
   - Can also be configured using `google.cloud.project` system property.
-  - If this option is not configured, the extension would infer GCP Project ID from the application default credentials. For more information on application default credentials, see [here](https://cloud.google.com/docs/authentication/application-default-credentials).
+  - This is a required option, the agent configuration will fail if this option is not set.
+- `GOOGLE_CLOUD_QUOTA_PROJECT`: Environment variable that represents the Google Cloud Quota Project ID which will be charged for the GCP API usage. To learn more about a *quota project*, see [here](https://cloud.google.com/docs/quotas/quota-project).
+  - Can also be configured using `google.cloud.quota.project` system property.
+  - If this option is not configured, the extension would infer GCP Quota Project ID from the application default credentials. For more information on application default credentials, see [here](https://cloud.google.com/docs/authentication/application-default-credentials).
```
Comment (aabmass) [inline gcp-auth-extension/README.md:42]: nit the extension doesn't infer the quota project id right? It just omits it to get the default behavior.

Comment (psx95) [reply gcp-auth-extension/README.md:42]: If not configured (set by the user), the quota project ID from the application default credentials is used as fallback.

If the inferred/resolved value of the quota project ID is null or empty, the header is omitted.

Comment (aabmass) [reply gcp-auth-extension/README.md:42]: OK I think my confusion is "infer" to me implies that the extension will always generate some kind quota project from ADC (not let it be null). I assumed it would try to set with ADC's project ID and always send

I think something like "it will use quota project found with ADC if available" and mention that this config is not required.

**Review by aabmass — APPROVED** [review]

File: gcp-auth-extension/README.md (hunk 1 above)
Comment (psx95) [reply gcp-auth-extension/README.md:42]: I will update the wording to be more clear.

File: gcp-auth-extension/src/test/java/io/opentelemetry/contrib/gcp/auth/GcpAuthAutoConfigurationCustomizerProviderTest.java
Comment (psx95) [reply gcp-auth-extension/src/test/java/io/opentelemetry/contrib/gcp/auth/GcpAuthAutoConfigurationCustomizerProviderTest.java:1]: I have added tests to ensure the expected behavior in regards to resolving the Quota project ID.

**Review by dashpole — APPROVED** [review]

**Review by trask — APPROVED** [review]
